### PR TITLE
#2434 - Function to remove linear dependencies in zonotope generators

### LINKED ISSUE #2434: Function to remove linear dependencies in zonotope generators
This should also be (optionally?) applied in `reduce_order`.

X-ref: #1453

### PR DESCRIPTION
**Pull request by schillic**
Closes #2434.

* [x] docs

### CODE REVIEW

File: src/Arrays/vector_operations.jl [hunk 1]
```diff
@@ -90,8 +91,47 @@ julia> samedir([1, 2, 3], [-1, -2, -3])
...
-function samedir(u::AbstractVector{N},
-                 v::AbstractVector{N}) where {N<:Real}
+function samedir(u::AbstractVector{<:Real}, v::AbstractVector{<:Real})
+    return _ismultiple(u, v; allow_negative=false)
+end
+
+"""
+    ismultiple(u::AbstractVector{<:Real}, v::AbstractVector{<:Real})
+
+Check whether two vectors point are linearly dependent.
```
Comment (mforets) [inline src/Arrays/vector_operations.jl:101]:
```suggestion
Check whether two vectors are linearly dependent.
```

File: src/Arrays/vector_operations.jl [hunk 2]
```diff
@@ -90,8 +91,47 @@ julia> samedir([1, 2, 3], [-1, -2, -3])
...
+
+### Input
+
+- `u` -- first vector
+- `v` -- second vector
+
+### Output
+
+`(true, k)` iff the vectors are identical up to a scaling factor `k ≠ 0`, and
```
Comment (mforets) [inline src/Arrays/vector_operations.jl]: shall we specify that it is `u = kv` or `v = ku`

File: src/Sets/Zonotope.jl [hunk 3]
```diff
@@ -636,3 +637,76 @@ function _vertices_list_2D_order_one(c::VN, G::MN; apply_convex_hull::Bool) wher
...
+                end
+                done[j2] = true
+                deleted = true
+            end
+        end
+        push!(G_new, gj1)
+    end
+
+    if deleted
+        G_new = hcat(G_new...)  # convert list of column vectors to matrix
```
Comment (mforets) [inline src/Sets/Zonotope.jl:702]:
```suggestion
        G_new = reduce(hcat, G_new)  # convert list of column vectors to matrix
```
i didn't check but reducing is usually more efficient than splatting

Comment (mforets) [reply src/Sets/Zonotope.jl:702]: ( unless `G_new` is a static array )

**Review by mforets — APPROVED** [review]

File: src/Arrays/vector_operations.jl (hunk 2 above)
Comment (schillic) [reply src/Arrays/vector_operations.jl]: yep, done

**Pull request merged**